pkg/ruler/base/storage.go

```diff
@@ -65,6 +67,9 @@ func (cfg *RuleStoreConfig) Validate() error {
 	if err := cfg.S3.Validate(); err != nil {
 		return errors.Wrap(err, "invalid S3 Storage config")
 	}
+	if err := cfg.BOS.Validate(); err != nil {
```

Comment:
this function always returns `nil`, why bother calling it here and implenting it as a no-op. We could follow a similar patter with the GCSConfig where we don't call `Validate` here and therefore don't need to implement it as a no-op. WDYT?

Reply:
I delete the function `Validate()` .You are right.It no need used now.